ingestion_tools/scripts/data_validation/tests/annotation/test_segmentationmask_annotation.py

```diff
@@ -0,0 +1,168 @@
+from typing import Dict
+
+import numpy as np
+import pytest
+from mrcfile.mrcinterpreter import MrcInterpreter
+from tests.helper_mrc import HelperTestMRCHeader
+
+ZATTRS_AXIS_ORDER = ["z", "y", "x"]
+
+
+# By setting this scope to session, scope="session" fixtures will be reinitialized for each run + voxel_spacing combination
+@pytest.mark.annotation
+@pytest.mark.parametrize("run_name, voxel_spacing", pytest.run_spacing_combinations, scope="session")
+class TestSegmentationMask(HelperTestMRCHeader):
+
+    @pytest.fixture(autouse=True)
+    def set_class_variables(
+        self,
+        seg_mask_annotation_mrc_headers: Dict[str, MrcInterpreter],
+        voxel_spacing: float,
+        canonical_tomogram_metadata: Dict,
+    ):
+        self.spacegroup = 1  # single 3D volume
+        self.mrc_headers = seg_mask_annotation_mrc_headers
+
+    ### BEGIN ZARR Self-consistency tests ###
+    def zarr_header_helper(
+        self,
+        seg_mask_annotation_zarr_headers: Dict[str, Dict[str, Dict]],
+        check_func: callable,
+        **kwargs,
+    ):
+        """Helper function to check the header of the zarr file."""
+        for zarr_filename, header_data in seg_mask_annotation_zarr_headers.items():
+            print(f"Checking {zarr_filename}")
+            check_func(header_data, zarr_filename, **kwargs)
+
+    def test_zarr_zattrs_path(
+        self,
+        seg_mask_annotation_zarr_headers: Dict[str, Dict[str, Dict]],
+    ):
+        """Check that the path is correct for a zarr annotation file."""
+
+        def check_zattrs_path(header_data, _zarr_filename):
+            del _zarr_filename
+            for i in range(3):
+                assert header_data["zattrs"]["multiscales"][0]["datasets"][i]["path"] == str(i)
+
+        self.zarr_header_helper(seg_mask_annotation_zarr_headers, check_zattrs_path)
+
+    def test_zarr_zattrs_voxel_spacings(
+        self,
+        seg_mask_annotation_zarr_headers: Dict[str, Dict[str, Dict]],
+        voxel_spacing,
+    ):
+        """Check that the voxel spacings are correct for a zarr annotation file."""
+
+        def check_zattrs_voxel_spacings(header_data, _zarr_filename, voxel_spacing):
+            del _zarr_filename
+            for i in range(3):
+                datasets_entry = header_data["zattrs"]["multiscales"][0]["datasets"][i]
+                assert datasets_entry["coordinateTransformations"][0]["scale"] == [voxel_spacing * (2**i)] * 3
+
```

Comment:
the scale, which is a list of length 3, should equal the voxel_spacing (scaled with the current binning folder which is 2^(binning factor), all repeated 3 times in a list